packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -113,7 +114,7 @@ describe("BnsConnection", () => {
     const nonce = await connection.getNonce({ pubkey: identity.pubkey });
     const signed = await profile.signTransaction(firstWalletId, identity, sendTx, bnsCodec, nonce);
     const response = await connection.postTx(bnsCodec.bytesToPost(signed));
-    await response.blockInfo.waitFor(info => info.state === BcpTransactionState.InBlock);
+    await response.blockInfo.waitFor(info => info.state !== TransactionState.Pending);
```

Comment:
This is a very common idiom, maybe expose this as a helper function (like you do with `isFailedTransaction`).

`function transactionProcessed(info: BlockInfo): boolean { return info.state !== TransactionState.Pending }`

It's not so much for the typing, but to avoid people using other checks (like `info => info.state === TransactionState.Success`) when they want this. They can use the other checks when they really only want the other case.

Reply:
Hmm, the helper function already exists in the wallet and I think we should be very careful with the amount of APIs we expose.

I wanted to suggest type checkers `isBlockInfoPending`,  `isBlockInfoSucceeded`, `isBlockInfoFailed` like we have for many union types. However, we can only use them directly if we expect success:

```ts
// ignore failures
await response.blockInfo.waitFor(isBlockInfoSucceeded)

// what we want
await response.blockInfo.waitFor(info => !isBlockInfoPending(info));
```

The last one is probably not what you desire, but better than what we have using a very generic API.

Reply:
I am fine with not exposing more and I will have fun in the client.
My major issue is that many tests use success/fail checks that may hang forever when the test would otherwise fail. You could make `transactionProcessed` unexported and use it in tests. But some way that broken tests report good errors.

Reply:
> My major issue is that many tests use success/fail checks that

Right, I will get rid of all `=== TransactionState.Success` in favour of `!== TransactionState.Pending` and check type explicitly